src/e2e-test/java/io/cdap/plugin/common/stepsdesign/CorePlugin.java

```diff
@@ -0,0 +1,134 @@
+    String gcsTargetBucket =  PluginPropertyUtils.pluginProp("gcsTargetBucket");
+    try {
+      // The output gcs folder will be like:
+      // e2e-test-[uuid]
+      // --2022-06-26-00-27/
+      // ----_SUCCESS
+      // ----part-r-0000
+      // ----part-r-0001
+      // ----part-r-...
+      // The number of part-r-* files should match the expected partitions.
+      int partitions = 0;
+      List<String> lst = new ArrayList<>();
+      for (Blob blob : StorageClient.listObjects(gcsTargetBucket).iterateAll()) {
+        String name = blob.getName();
+        if (name.contains("part-r")) {
+          partitions++;
+          try (InputStream inputStream = new ByteArrayInputStream(blob.getContent())) {
+            readInputStream(inputStream, lst);
+          }
+        }
+      }
+      Path path = Paths.get(Objects.requireNonNull(CorePlugin.class.getResource
+        ("/" + PluginPropertyUtils.pluginProp(file))).getPath()).toAbsolutePath();
+      if (Integer.parseInt(expectedPartitions) != 0) {
+        Assert.assertEquals("Output partition should match",
+                            partitions, Integer.parseInt(PluginPropertyUtils.pluginProp(expectedPartitions)));
+      }
+      Assert.assertTrue("Output content should match",
+                        Strings.equals(getSortedCSVContent(lst), new String(Files.readAllBytes(path))));
+    } catch (StorageException | IOException e) {
+      if (e.getMessage().contains("The specified bucket does not exist")) {
+        BeforeActions.scenario.write("GCS Bucket " + gcsTargetBucket + " does not exist.");
+      } else {
+        Assert.fail(e.getMessage());
+      }
+    }
+  }
+  private String getSortedCSVContent(List<String> lst) {
+    // Since the spark output files aren't guaranteed to be ordered, the output entry with [id, field1, field2, ....]
+    // schema needs to be sorted by id for comparison purpose.
+    lst.sort((s1, s2) -> {
+      String id1 = s1.split(",")[0], id2 = s2.split(",")[0];
+      return Integer.parseInt(id1) - Integer.parseInt(id2);
+    });
+    StringBuilder sb = new StringBuilder();
+    for (String s : lst) {
+      sb.append(s);
+    }
+    return sb.toString();
+  }
+
+  private void readInputStream(InputStream input, List<String> lst) throws IOException {
+    try (BufferedReader reader = new BufferedReader(new InputStreamReader(input))) {
+      String line;
+      while ((line = reader.readLine()) != null) {
+        lst.add(line + "\n");
+      }
+    }
+  }
+
+  @Then("Validate output file generated by file sink plugin is equal to expected output file {string}")
+  public void validateOutputFileGeneratedByFileSinkPluginIsEqualToExpectedOutputFile(String expectedOutputFilePath)
+    throws IOException {
+    Optional<Path> sinkOutputFile =  Files.walk(Paths.get(PluginPropertyUtils.pluginProp("filePluginOutputFolder")))
+      .filter(Files::isRegularFile).filter(file -> file.toFile().getName().startsWith("part-r")).findFirst();
```

Comment:
read all part files, not just the first one

Reply:
We are expecting only single part file with provided input file... thats why checking only first file.

Reply:
The test code should be written in a generic way - it should work correctly with any input file.

Reply:
Removed the `CorePlugin.java` file.